src/code/z_viszbuf.c

```diff
@@ -0,0 +1,112 @@
+/**
+ * @file z_viszbuf.c
+ * Description: Visualise the z-buffer (AKA depth buffer), using cycling RGBA or a single fading color.
+ *
+ * This is done by reading the z-buffer as if it were a color image, the format of which is specified by `useRgba`:
+ *
+ * - false will produce a monotonic fade from primColor to envColor as depth increases
+ * - true produces vibrant almost-periodic-looking bands.
+ *
+ * In both cases this occurs because of the format the depth information takes: it is 18-bit, and is a nonnegative
+ * floating-point number with
+ *
+ *     bbb      mmmmmmmmmmm dd|dd
+ *     exponent mantissa    dz value (only first 16 bits visible to CPU)
+ *
+ * Reading z-buffer as IA16:
+ *
+ *     bbbmmmmm mmmmmmdd
+ *     iiiiiiii aaaaaaaa
+ *
+ * Since floating-point numbers of this format have the same ordering as their binary/hex representation, increasing
+ * the depth also increases the intensity in the IA16 representation and hence the interpolation parameter used to
+ * combine primColor and envColor. The alpha is ignored by the RenderMode.
+ *
+ * Reading z-buffer as RGBA16:
+ *
+ *     bbbmm mmmmm mmmmd d
+ *     rrrrr ggggg bbbbb a
+ *
+ * The red increases monotonically with the depth. The significant visible oscillation is the green component, because
+ * it rolls over every time the second-most-significant bit of the mantissa increments. The blue component oscillates
+ * too rapidly to be particularly visible (it rolls over when the 7th-most-significant bit increments). The alpha is
+ * again ignored by the RenderMode.
+ */
+
+#include "global.h"
+
+// Height of the fragments the z-buffer is split into.
+// It is the maximum amount of lines such that all rgba16 SCREEN_WIDTH-long lines fit into TMEM.
+#define VISZBUF_ZBUFFRAG_HEIGHT (TMEM_SIZE / (SCREEN_WIDTH * G_IM_SIZ_16b_BYTES))
+
+// z-buffer
+extern u16 D_0E000000[];
+
+/**
+ * Initialise to IA type with white and black as default colours.
+ */
+void VisZbuf_Init(VisZbuf* this) {
+    this->useRgba = false;
+    this->setScissor = false;
+    this->primColor.r = 255;
+    this->primColor.g = 255;
+    this->primColor.b = 255;
+    this->primColor.a = 255;
+    this->envColor.a = 255;
+    this->envColor.r = 0;
+    this->envColor.g = 0;
+    this->envColor.b = 0;
+}
+
+void VisZbuf_Destroy(VisZbuf* this) {
+}
+
+void VisZbuf_Draw(VisZbuf* this, Gfx** gfxp) {
+    Gfx* gfx = *gfxp;
+    u16* zbufFrag = D_0E000000;
+    s32 fmt;// = (this->useRgba == false) ? G_IM_FMT_IA : G_IM_FMT_RGBA;
```

Comment:
What's the reasoning behind not using the ternary or at least the `== false` here?
Also this looks like it may need a formatter run?